src/include/distributed/security_utils.h

```diff
@@ -0,0 +1,26 @@
+/*-------------------------------------------------------------------------
+ *
+ * security_utils.h
+ *   security related utility functions.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef SECURITY_UTILS_H_
+#define SECURITY_UTILS_H_
+
+#include "postgres.h"
+#include "miscadmin.h"
+
+#define PushCitusSecurityContext() \
+	Oid savedUserId_DONTUSE = InvalidOid; \
+	int savedSecurityContext_DONTUSE = 0; \
+	GetUserIdAndSecContext(&savedUserId_DONTUSE, &savedSecurityContext_DONTUSE); \
```

Comment:
this seems a bit too lazy to me :)

I'm neutral to replacing these calls, but I would expect the "PushCitusSecurityContext" function to push the user ID a stack, such that you don't necessarily need to use push and pop in the same block. There's no good reason for it to be a macro.